### ISSUE TITLE
Could not load file or assembly 'Newtonsoft.Json, Version=9.0.0.0

### ISSUE THREAD

**Issue by Grime121** [PureStorage-Connect/PowerShellSDK2#24]
I upgraded to the latest version of the PS module (2.26.70), and I am now getting the following error when I try to connect to my arrays. I removed the module entirely and installed it from scratch, but I am still getting the same error. I don't know what version I had installed previously, but I did not get this error with it. I think it had a '7' in the version number, so maybe it was 2.2.272 or 2.2.276.

Connect-Pfa2Array : Could not load file or assembly 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. 
The system cannot find the file specified.

I checked the module's folder contents, and I see the Newtonsoft.Json.dll file in there. It says it is version 13, though.

**Comment by mikenelson-io** [comment]
Does this resolve?
https://support.purestorage.com/Solutions/Microsoft_Platform_Guide/a_Windows_PowerShell/KB%3A_OAuth_authentication_throws_error_for_Newtonsoft.dll_when_using_Connect-Pfa2Array_cmdlet

**Comment by Grime121** [comment]
I didn't install the Pure PS module using the MSI. I used the Install-Module method. I still would be open to trying this fix though if I could be certain it would not cause issues with any other software that is installed on this server. This server also has Veeam installed on it, which uses Newtonsoft.Json. It also has some Ivanti software installed on it that appears to use Newtonsoft.Json. I'm not sure what else might be installed on it that is using Newtonsoft.Json, but considering there are about 50+ instances of that DLL on the system, I'm concerned I might break something else if I modify the machine.config file. What do you think?

**Comment by Grime121** [comment]
I tried making the change suggested in that article, but it broke two of the Ivanti services when I did (they wouldn't start). I was however able to modify the change a bit to get it working. I changed the version range from "0.0.0.0-12.0.0.0" to "0.0.0.0-11.0.0.0".

I'm actually not quite sure why that was necessary since the Newtonsoft.Json.dll files the two services use are version 12.0.3.23909 (you'd think that wouldn't be included in "0.0.0.0-12.0.0.0"), but it was. Everything appears to be working after changing that range, though.
